core/tests/protractor/privileges.js

```diff
@@ -46,10 +46,18 @@ describe('Permissions for private explorations', function() {
 
       users.login('eve@privileges.com');
       general.openEditor(explorationId);
+
       // Eve is redirected to the homepage.
-      expect(browser.getCurrentUrl()).toEqual(
-        general.SERVER_URL_PREFIX + '/dashboard');
-      users.logout();
+      browser.getCurrentUrl().then(function(curUrl) {
+        var dashboardURL = general.SERVER_URL_PREFIX + '/dashboard';
+        var libraryURL = general.SERVER_URL_PREFIX + '/library';
+
+        // Homepage can be either dashboard or library dependin on learner
+        // Status
+        var atHomeFlag = (curUrl === libraryURL) || (curUrl === dashboardURL);
```

Comment:
In this test, from the source code, I know eve didn't contribute to any exploration so this means that I should check if the eve is on library page ? Or shall I test both behavior by creating a separate contribution for eve ?